crates/proof/proof/src/lib.rs

```diff
@@ -3,7 +3,7 @@
 #![deny(unused_must_use, rust_2018_idioms)]
 #![cfg_attr(docsrs, feature(doc_cfg, doc_auto_cfg))]
 #![cfg_attr(not(test), warn(unused_crate_dependencies))]
-#![no_std]
+#![cfg_attr(not(test), no_std)]
```

Comment:
I'd prefer we not do this if we can avoid it.

If we need `std` in tests, just make the test itself as `#[cfg(feature = "std")]` or whatever feature flag allows for `std`. Here we have an `std` feature flag for `kona-proof` so i think we can use that.

Reply:
The reason why is this leaks `std` into all tests including `no_std` tests. It loosens the testing environment constraints.